embassy-stm32/src/eth/v2/mod.rs

```diff
@@ -35,11 +35,11 @@ impl interrupt::typelevel::Handler<interrupt::typelevel::ETH> for InterruptHandl
 }
 
 /// Ethernet driver.
-pub struct Ethernet<'d, T: Instance, P: PHY> {
+pub struct Ethernet<'d, T: Instance, P: PHY, const NUM_PINS: usize> {
```

Comment:
we should strongly avoid adding generic params, even if it requires doing things at runtime. 

There's a few options. Use `heapless::Vec` with a max capacity of 14, or do an enum:

```rust
enum Pins<'d> {
   Rmii([PeripheralRef<'d, AnyPin>; 9]),
   Mii([PeripheralRef<'d, AnyPin>; 14]),
}

Reply:
Thank you, I really like the idea with the enum :slightly_smiling_face: